DTAConfig/OptionPanels/ComponentsPanel.cs

```diff
@@ -134,30 +151,33 @@ private void Btn_LeftClick(object sender, EventArgs e)
                 if (cc.LocalIdentifier == cc.RemoteIdentifier)
                 {
                     File.Delete(ProgramConstants.GamePath + cc.LocalPath);
-                    btn.Text = "Install";
+                    btn.Text = "Install (" + GetSizeString(cc.RemoteSize) + ")";
                     return;
                 }
 
                 btn.AllowClick = false;
 
                 cc.DownloadFinished += cc_DownloadFinished;
                 cc.DownloadProgressChanged += cc_DownloadProgressChanged;
-                Thread thread = new Thread(cc.DownloadComponent);
-                thread.Start();
+                cc.DownloadComponent();
             }
             else
-            {
+            {   
+                string archiveSizeMsg = "";
+                if (cc.Archived && cc.RemoteArchiveSize != cc.RemoteSize)
+                    archiveSizeMsg = " (size of the download is " + GetSizeString(cc.RemoteArchiveSize) + ")";     
+                  
                 var msgBox = new XNAMessageBox(WindowManager, "Confirmation Required".L10N("UI:DTAConfig:UpdateConfirmRequiredTitle"),
                     string.Format(("To enable {0} the Client will download the necessary files to your game directory." +
-                    Environment.NewLine + Environment.NewLine + "This will take an additional {1} of disk space, and the download may last" +
+                    Environment.NewLine + Environment.NewLine + "This will take an additional {1} of disk space{2}, and the download may last" +
```

Comment:
That `{2}` looks a bit wrong, maybe spaces at least are needed? I don't get why it's placed here.

Reply:
It is replaced by `archiveSizeMsg`, which is conditionally either a proper message or an empty string, hence the lack of space there (the message itself is prefixed with one if it is set.